### ISSUE TITLE
[Feature Request]: Organize apps by keyboard shortcuts or in preferences pane

### ISSUE THREAD

**Issue by RDK-dev** [jordanbaird/Ice#83]
### Description

I'm a screen reader (VoiceOver) user. Therefore, it is almost impossible for me to drag and drop icons in the menubar to hide them. Would it be possible to organize apps—hidden or not—by keyboard shortcuts or by having a dedicated pane in preferences where you can select which apps are hidden, always hidden, etc.?

**Comment by jordanbaird** [comment]
Yes, I can work on this. The app really could use better accessibility all around, so this can come with that.

**Comment by davidsule** [comment]
@jordanbaird I also really support being able to arrange icons in the settings, it would be super useful for everyone, not just from accessibility point of view. If Ice would remember the icons the user had previously even if they are not in the menu bar currently, the user could establish an absolute order between them so that no matter what combination of icons are present the oder is always defined (+ maybe the location of previously unseen icons could also be set).

(I saw you wrote under another issue that Ice does not currently handle the order itself, so I guess this is all contingent on that first).

**Comment by FlickerSoul** [comment]
@jordanbaird I migrated from bartender to this awesome project couple of days ago and wonder if I can contribute to the codebase on this feature (organizing items in settings and the ability to use shortcuts)? I have some Swift experience and it would take me a week or two to get started with the existing code. Does that sound like a good idea?

**Comment by jordanbaird** [comment]
@FlickerSoul I would love the help! I would like to do the organization-in-settings feature, as I want to integrate it with profiles, and I already have a design for that, but I'm more than happy to turn the organization-with-shortcuts feature over to you. Do you have any ideas on how you might present the feature in the interface? I can help with some ideas if you need.

**Comment by FlickerSoul** [comment]
@jordanbaird Thank you so much for the quick reply! 

I'm not entirely clear at this moment how to do the organization-with-shortcuts solely on menu bar. Does macOS allow focus on Ice menu bar? If so, it seems natural that users can toggle edit mode, which shifts focus on Ice menu bar, select certain icon by clicking or arrow keys, and then use some defined hotkey (`cmd + arrow` to reorder, `cmd + h` to toggle hide, etc.) arrange icons. What do you think? 

I had envisioned the shortcuts being integrated with the organization-in-settings feature, as this already allows users to select and arrange icons. Adding shortcuts in this context seems straightforward.

Speaking of shortcuts, I also considered mimicking Bartender's functionality, allowing users to assign specific shortcuts to show or click an icon, but this seems like a separate feature.

**Comment by jordanbaird** [comment]
@heyRDK Just checking in -- do Ice's current features satisfy your needs? Mainly with the drag-and-drop interface in settings. I'm still completely open to implementing some form of keyboard-based item arranging, I've just been very busy with other features and bugs. If the current features do work for you, I think it would be good if we closed this and opened a separate, more focused issue for the keyboard arranging.

@FlickerSoul Sorry for getting back to you so late on this. If you're still interested in implementing the keyboard-based item arranging, I can help guide you in the right direction based on the current menu bar layout implementation.

**Comment by heyRDK** [comment]
I updated to the latest version. When I open Settings > Menu Bar Items, VoiceOver announces 'empty scroll area' below Visible, Hidden, and Always-Hidden. Should the apps be available in those sections, or do I need to drag and drop them from the menu bar to different sections?

**Comment by FlickerSoul** [comment]
@jordanbaird No worries at all! I am still interested in implementing this feature and would love to receive some guidance! Thanks! :)

**Comment by heyRDK** [comment]
@jordanbaird Now it is possible to drag and drop items in settings. I’m running version 0.10.5.

One thing, when you navigate the list of apps in the settings, VoiceOver says: «layout item». However, when I invoke help menu on each object by pressing VO key (⌃⌥) + ⇧ + H, I get to know the name of the app.

Is it possible for you to swap those two, that when you browsing the list, VoiceOver should announce app names instead of layout item?
